### PR TITLE
fix(BaseClient): produce an error for non object options

### PR DESCRIPTION
**Pull request by J-Human**
**Please describe the changes this PR makes and why it should be merged:**

This throws an error to the user if the `options` provided is not an object. It prevents more confusing and misleading errors given by the process. For example, one may have something like this:
```ts
new WebhookClient('id', 'token');
```
Which is, in fact, an outdated way of creating Webhooks. I wouldn't say we *shouldn't* place it because it's the user's fault, though. Since the options parameter is a string, it cannot append the default arguments, hence producing a misleading error:
`cannot append property shardCount on string "..."`
I've seen lots of users in the server that have errors like the above. 

Furthermore, this prevents using the wrong data in other parts of client-related things. Some users don't have a proper IDE, hence they won't receive any feedback from the typings either, this does it for the interpreter.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Comment by Keke27210** [comment]
Documentation is pretty clear on how to build each class, if people don't want to be bothered to check it, it's their problem.
discord.js is a beginner-friendly library (in spite of itself? e.g. the stricter Snowflake type) but its source code should not be containing extra-(useless)-code.

JavaScript being weakly typed is one of its forces, but also one of its drawbacks (which is why TypeScript was born), but imo it's not up to the library to check for those inputs. If people want to code "safely", they can switch to TypeScript, get a good linter, or even better (but maybe a little bit too much) read documentation.

All of this to say I'm not really a big fan of that, as it would probably start up a trend of implementing type checks everywhere in the source code that'd end up taking a little bit of performance at runtime.

File: src/client/BaseClient.js
Comment (NotSugden) [inline src/client/BaseClient.js:17]: i think it would be cleaner to
```js
if (typeof options !== 'object' || options === null) {
  throw [...];
}

this.options = [...];
```

**Review by kyranet — APPROVED** [review]

File: src/client/BaseClient.js
Comment (kyranet) [inline src/client/BaseClient.js:18]:
```suggestion
      throw new TypeError('INVALID_TYPE', 'options', 'object', true);
```

File: src/client/BaseClient.js
Comment (kyranet) [inline src/client/BaseClient.js:4]:
```suggestion
const { TypeError } = require('../errors');
```

**Review by kyranet — CHANGES_REQUESTED** [review]
Just noticed something, the `'INVALID_TYPE'` error is used exclusively with `TypeError` across the library.

**Review by vladfrangu — APPROVED** [review]
LGTM, although we should decide if we validate all inputs, just some, or none.. personally a fan of runtime validation (you can see that in builders)

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Comment by iCrawl** [comment]
This needs a rebase.

**Comment by J-Human** [comment]
Done

**Review by iCrawl — APPROVED** [review]

**Pull request merged**
